tests/ui/transmute/transmute-different-sizes.stderr

```diff
@@ -4,44 +4,57 @@ error[E0512]: cannot transmute between types of different sizes, or dependently-
 LL |     let _: i8 = transmute(16i16);
    |                 ^^^^^^^^^
    |
-   = note: source type: `i16` (N bits)
-   = note: target type: `i8` (N bits)
+   = note: source type: `i16`
+           the type has N bits
+   = note: target type: `i8`
+           the type has N bits
```

Comment:
Please revert these diagnostic changes. They are orthogonal to your fixes and I'm not sure they should be done at all

Reply:
By reverting the following errors:
https://github.com/rust-lang/rust/pull/135158/files#diff-0bb9b9dce5bc860ae020a408724e5ac4ec19803287063e92ac447928c12768fbR41-R45
the error message would look something like this:
```rust
= note: source type: `*mut T` (pointer to `T`)
= note: target type: `PtrAndOverAlignZST<T>` (the type `<T as Pointee>::Metadata` is too generic to determine its layout. the size of this type is unknown at compile-time)
```

IMO, the first note is actually quite clear. However, the new error introduced in this PR, shown in the second note, is written on a single line and might become too long (especially when the terminal does not wrap text).
Should I try to shorten the new error? Or do you have any other suggestions?

Reply:
yea I think just choosing a shorter (single line) message is best. 

you can leave out the `the size of this type is unknown at compile-time` entirely

Maybe just 

```
(`<T as Pointee>::Metadata` is too generic)
```

and then we can figure out how to add additional notes to those messages in general, as I'm sure others would like more details, too

Reply:
Ok, thanks for your feedback! 

I agree to remove `the size of this type is unknown at compile-time`

I would like to keep `(the layout of <T as Pointee>::Metadata is too generic)` since we know it would generate a problem in determining the layout of the type.

> and then we can figure out how to add additional notes to those messages in general, as I'm sure others would like more details, too

Agreed, I avoid handling it in this PR